File: Java/Bank_2/src/bank_2/Banco.java

package bank_2;

import java.util.ArrayList;
import java.util.Scanner;

public class Banco 
{
    Empleado empleado;

    Scanner leer = new Scanner(System.in);
    private ArrayList<Empleado> empleados;

    public Banco() {
        empleados = new ArrayList<>();
    }

    public void agregarEmpleado(Empleado empleado) {
        empleados.add(empleado);
    }

    public void mostrarEmpleados()
    {
        if (empleados.isEmpty()) 
        {
            System.out.println("\n***No hay empleados registrados***");
            return;
        }
        System.out.println("\n*** LISTA DE EMPLEADOS ***");
        for (Empleado empleado : empleados) 
        {
            empleado.verCuentas();
        }
    }

    public void mostrarEmpleado(String nombre) 
    {
    boolean encontrado = false;
    for (Empleado empleado : empleados) {
        if (empleado.getName().equalsIgnoreCase(nombre)) 
        {
            empleado.verCuentas();
            encontrado = true;
            break;
        }
    }
    if (!encontrado) 
    {
        System.out.println("***Empleado no encontrado***");
    }
}

    public Empleado buscarEmpleado(String nombre) {
    for (Empleado empleado : empleados) {
        if (empleado.getName().equalsIgnoreCase(nombre)) {
            return empleado;
        }
    }
    return null;
}
}

File: Python/Banco/Banco.py


class Banco:
    def __init__(self):
        self.empleados = []

    def agregar_empleado(self, empleado):
        self.empleados.append(empleado)

    def mostrar_empleados(self):
        if not self.empleados:
            print("\n***No hay empleados registrados***")
            return
        print("\n*** LISTA DE EMPLEADOS ***")
        for empleado in self.empleados:
            empleado.ver_cuentas()

    def mostrar_empleado(self, nombre):
        encontrado = False
        for empleado in self.empleados:
            if empleado.name.lower() == nombre.lower():
                empleado.ver_cuentas()
                encontrado = True
                break
        if not encontrado:
            print("***Empleado no encontrado***")

    def buscar_empleado(self, nombre):
        for empleado in self.empleados:
            if empleado.name.lower() == nombre.lower():
                return empleado
        return None
